File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):

    clothes_dict = {}
    for item, item_type in clothes:
        if item_type not in clothes_dict.keys():
            clothes_dict[item_type] =1
        else:
            clothes_dict[item_type] +=1
    
    answer = 1
    for cnt in clothes_dict.values():
        answer*=(cnt+1)
        
    return answer -1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    let cMap = new Map();
    
    for(let [c, k] of clothes) {
        if(cMap.has(k)) cMap.set(k, cMap.get(k) + 1);
        else cMap.set(k, 1);
    }
    
    for(let i of cMap.values()) {
        answer *= (i+1)
    }
    
    answer -= 1;
    return answer;
}